第2章終了

shmn7iii opened the pull request:
成果物：https://morning-tor-81081.herokuapp.com/

shmn7iii committed:
add toy_app

shmn7iii committed:
add hello render

shmn7iii committed:
create toy_app

shmn7iii committed:
bundle _2.2.17_ lock --add-platform x86_64-linux

for heroku

shmn7iii committed:
generate user model

shmn7iii committed:
generate micropost model

shmn7iii committed:
add content's maximum length on microposts

shmn7iii committed:
add relationship between user and micropost

shmn7iii committed:
delete .keeps

最初から生成されないようにしたいんだけど...

shmn7iii committed:
update readme

shmn7iii reviewed (commented):
**[自分メモ] .keepファイルについて**
`rails new` 時に `–skip-keeps` オプション追加で生成されないようにできる
参考：https://railsdoc.com/page/rails_new

p0pemaru commented on README.md:
```diff
@@ -1,102 +1,20 @@
-Heroku: https://gentle-shore-22479.herokuapp.com/
 
-## 備考
+## 学び
```
このセクションとても良いと思います！
前章の内容や環境構築メモを上書きしている形で更新していますが、消さずにどんどん追記していく形で問題無いと思いますよ 👍

p0pemaru commented on app/assets/stylesheets/microposts.scss:
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Microposts controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
自動生成されるファイルですが、使用しないと思うので削除してもらったほうが見通しが良くなります。

p0pemaru commented on app/assets/stylesheets/users.scss:
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Users controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
これも不要ですね！

p0pemaru commented on app/controllers/microposts_controller.rb:
```diff
@@ -0,0 +1,74 @@
+class MicropostsController < ApplicationController
+  before_action :set_micropost, only: [:show, :edit, :update, :destroy]
```
```suggestion
  before_action :set_micropost, only: %i[show edit update destroy]
```
nits.
シンボルの配列は%記法を用いることが多いです。

p0pemaru commented on app/controllers/microposts_controller.rb:
```diff
@@ -0,0 +1,74 @@
+  end
+
+  private
+    # Use callbacks to share common setup or constraints between actions.
+    def set_micropost
+      @micropost = Micropost.find(params[:id])
+    end
+
+    # Never trust parameters from the scary internet, only allow the white list through.
+    def micropost_params
+      params.require(:micropost).permit(:content, :user_id)
+    end
```
```suggestion
  # Use callbacks to share common setup or constraints between actions.
  def set_micropost
    @micropost = Micropost.find(params[:id])
  end

  # Never trust parameters from the scary internet, only allow the white list through.
  def micropost_params
    params.require(:micropost).permit(:content, :user_id)
  end
```

private メソッドの定義は`private`行のインデントと揃えるのが一般的です。

p0pemaru commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,74 @@
+  end
+
+  private
+    # Use callbacks to share common setup or constraints between actions.
+    def set_user
+      @user = User.find(params[:id])
+    end
+
+    # Never trust parameters from the scary internet, only allow the white list through.
+    def user_params
+      params.require(:user).permit(:name, :email)
+    end
```
ここのインデントも↑の指摘と同じですね！

p0pemaru commented on app/helpers/microposts_helper.rb:
```diff
@@ -0,0 +1,2 @@
+module MicropostsHelper
+end
```
これも削除して問題ないですね。

p0pemaru commented on app/helpers/users_helper.rb:
```diff
@@ -0,0 +1,2 @@
+module UsersHelper
+end
```
これも不要ですね〜

p0pemaru commented on app/views/layouts/application.html.erb:
```diff
@@ -1,7 +1,7 @@
 <!DOCTYPE html>
 <html>
   <head>
-    <title>HelloApp</title>
+    <title>ToyApp</title>
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
 
```
FYI
rails のデフォルトのテンプレートエンジンは erb ですが、よりスッキリと html を記載できる slim が代わりによく使用されます。
https://qiita.com/ngron/items/c03e68642c2ab77e7283

rails tutorial では erb のまま進めてもらっても問題無いので、こういったテンプレートもあるということだけ覚えておいてください :+1:

ちなみに今回の部分であれば次のように slim では書けます。
```slim
doctype html
html
  head
    title ToyApp
    = csrf_meta_tags
    = csp_meta_tag
```

p0pemaru commented on db/migrate/20220303055523_create_users.rb:
```diff
@@ -0,0 +1,10 @@
+class CreateUsers < ActiveRecord::Migration[6.0]
+  def change
+    create_table :users do |t|
+      t.string :name
```
FYI
実際のサービスではここで`null: false`などバリデーションを指定することが多いです。

p0pemaru commented on db/migrate/20220303061143_create_microposts.rb:
```diff
@@ -0,0 +1,10 @@
+  def change
+    create_table :microposts do |t|
+      t.text :content
+      t.integer :user_id
```
FYI
実際のサービスでは`t.references :user`などと書くことが多いです。
cf. https://qiita.com/ryouzi/items/2682e7e8a86fd2b1ae47

shmn7iii committed:
update readme

https://github.com/shmn7iii/rails_tutorial/pull/2#discussion_r818404090

shmn7iii committed:
delete auto generation files

https://github.com/shmn7iii/rails_tutorial/pull/2#discussion_r818406075
https://github.com/shmn7iii/rails_tutorial/pull/2#discussion_r818406344
https://github.com/shmn7iii/rails_tutorial/pull/2#discussion_r818417860
https://github.com/shmn7iii/rails_tutorial/pull/2#discussion_r818418067

shmn7iii committed:
change symbol's notation

https://github.com/shmn7iii/rails_tutorial/pull/2#discussion_r818414861

shmn7iii committed:
change indent in private method

https://github.com/shmn7iii/rails_tutorial/pull/2#discussion_r818416662
https://github.com/shmn7iii/rails_tutorial/pull/2#discussion_r818417310

shmn7iii committed:
update readme

shmn7iii committed:
update readme

shmn7iii committed:
null validation

https://github.com/shmn7iii/rails_tutorial/pull/2#discussion_r818433316

shmn7iii committed:
t.references :user

https://github.com/shmn7iii/rails_tutorial/pull/2#discussion_r818435908

shmn7iii committed:
update readme

shmn7iii commented:
@haseponpon さん
反映しました！

- 自動生成ファイル削除：https://github.com/shmn7iii/rails_tutorial/pull/2/commits/f2cd35809b87c18835f121764471c0a576745796

- シンボルの記法変更：https://github.com/shmn7iii/rails_tutorial/pull/2/commits/8f88ab8ad497a1a697eded08d34343d39e4647ae

- privateメソッドのインデント変更：https://github.com/shmn7iii/rails_tutorial/pull/2/commits/2dfe00eca3be988887ec5002befb16f186e7c6b0

- null: false追加：https://github.com/shmn7iii/rails_tutorial/pull/2/commits/252cdf24bf0e51ba0fadf4bd3c5463370f455437

- t.references :user に変更：https://github.com/shmn7iii/rails_tutorial/pull/2/commits/7f7835a3def35d76ac04950bfef0135b6a7d3be6


です！

p0pemaru reviewed (approved):
LGTM!

commit message で修正内容把握できるので、idは転記してもらわなくても大丈夫ですよ！
特に連絡事項や質問などがなければ
修正終了後 push→ re-request reviewボタンをポチる
だけで問題無いです:+1:

また今回オプションで tutorial に無いバリデーションも追加してもらってますが、万一今後の章を進めていく際にこれが起因でエラーが出た場合などには外してもらっても大丈夫です！

shmn7iii commented:
了解です！ありがとうございます！
